File: Lab5/Zadanie3.cpp
#include <iostream>
#include <locale.h>

using namespace std;

int S(int);

int main()
{
    setlocale(LC_CTYPE, "Polish");
    int n;
    cout << "Podaj n:" << endl;
    cin >> n;
    if (n >= 0)
    {
        cout << "Wartość wyrazu ciągu S dla n=" << n << " wynosi:" << S(n) << endl;
    }
    else
    {
        cout << "Podano złą wartość dla n." << endl;
    }
}

int S(int n)
{
    if (n == 0)
    {
        return 1;
    }
    else if(n == 1)
    {
        return 1;
    }
    else
    {
        int* T = new int[n + 1];
        T[0] = 1;
        T[1] = 1;
        for (int i = 2; i <= n; i++)
        {
            T[i] = 2 * T[i - 1] - T[i-2];
        }
        int rezultat = T[n];
        return rezultat;
    }
}

File: Lab5/zadanie3.py
def S(n):
    if n==0:
        return 1
    elif n==1:
        return 1
    ST = [1,1]
    i = 2
    while(i<=n):
        ST.append(2*ST[i-1]-ST[i-2])
        i+=1
    return ST[n]

n = int(input("Podaj n:"))
print(f"Wartość wyrazu dla {n} wynosi {S(n)}")